src/Map/Map.router.js

```diff
@@ -41,26 +39,67 @@ export class MapRouter {
     }
   }
 
+  async getAtlas(req) {
+    let atlas = []
+
+    let nw
+    let se
+    try {
+      nw = server.extractFromReq(req, 'nw')
+      se = server.extractFromReq(req, 'se')
+    } catch (_) {
+      // keep undefined
+    }
+
+    try {
+      const address = server.extractFromReq(req, 'address')
+      atlas = await Atlas.inRangeFromAddressPerspective(nw, se, address)
+    } catch (error) {
+      atlas = await Atlas.inRange(nw, se)
+    }
+
+    const map = {}
+
+    for (const row of atlas) {
```

Comment:
What do you think of moving this to a function? It can return a new object

```javascript
funtion normalizeRow(row) {
 const normalizedRow = {
         x: row.x,
         y: row.y,
         type: row.type
       }
       if (row.owner && [TYPES.taken, TYPES.onSale].includes(row.type)) {
         normalizedRow.owner = row.owner.slice(0, 6)
       }
       if (row.price) normalizedRow.price = row.price
       if (row.name) normalizedRow.name = row.name
       if (row.estate_id) normalizedRow.estate_id = row.estate_id
       if (row.is_connected_left) normalizedRow.left = 1
       if (row.is_connected_top) normalizedRow.top = 1
       if (row.is_connected_topleft) normalizedRow.topLeft = 1
     }

  return normalizedRow
}
```

coded on github